px-app-route/px-app-route-helpers.js

```diff
@@ -0,0 +1,113 @@
+(function() {
+  var PxAppRouteHelpers = {
+    isDeepEqual: function(a1, a2) {
+      if (!Array.isArray(a1) || !Array.isArray(a2)) return false;
+      if (a1.length !== a2.length) return false;
+
+      for (let i=0; i<a1.length; i++) {
+        if (a1[i] !== a2[i]) {
+          return false;
+        }
+      }
+      return true;
+    },
+
+    encodeAllPaths: function(routes, options) {
+      if (typeof routes !== 'object') return;
+
+      options = options || {};
+      var pathSegment = options.segmentPathsWith || '/';
+      var modelSegment = options.segmentModelsWith || '~';
+      var prefixFirstPath = typeof options.prefixFirstPath === 'boolean' ? options.prefixFirstPath : true;
+      var path = '';
+
+      if (Array.isArray(routes.navRoute) && routes.navRoute.length > 0) {
+        path += this.encodePath(routes.navRoute, pathSegment, prefixFirstPath);
+      }
+      if (Array.isArray(routes.assetRoute) && routes.assetRoute.length > 0) {
+        var assetPath = this.encodePath(routes.assetRoute, pathSegment, prefixFirstPath);
+        if (assetPath.length > 0) {
+          path += modelSegment + assetPath;
```

Comment:
If we do es6 this, would template literals clean this up any? Not sure of intent etc. but just sayin ;)